src/vm/prestub.cpp

```diff
@@ -1197,7 +1197,13 @@ PrepareCodeConfig::JitOptimizationTier PrepareCodeConfig::GetJitOptimizationTier
     _ASSERTE(methodDesc != nullptr);
     _ASSERTE(config == nullptr || methodDesc == config->GetMethodDesc());
 
-    if (config != nullptr)
+    // The code inside this block may take one or more locks. If process detach has already occurred, then other threads would
+    // have already been abruptly terminated, which may have left locks orphaned, so it would not be feasible to attempt taking
+    // a lock on process detach. This function is called from ETW-related code on shutdown where unfortunately it processes and
+    // sends rundown events upon process detach. If process detach has already occurred, then resort to best-effort behavior
+    // that may return inaccurate information. This is a temporary and targeted fix for a clear problem, and should not be used
+    // long-term.
+    if (config != nullptr && !g_fProcessDetach)
```

Comment:
Yes getting the correct info involves taking a lock, which can't be done when `g_fProcessDetach == true`, so the value returned here may end up being incorrect. For methods that were jitted while profiling, there would already have been an event soon after jitting with the correct info, so I believe they would not be affected, it would only send potentially incorrect info for methods that were already jitted prior to starting the profiler.

Reply:
Would it be possible to return an "Unknown" value or something along those lines when `g_fProcessDetach == true` so that it's clear when the data can be trusted and when it can't?  Other than that, this fix seems reasonable to me.

Reply:
Good idea, looks like we already have an `Unknown` value to support older runtimes, that would be appropriate to use here

Reply:
Great.  One follow-up question on this.  Because `g_fProcessDetach == true` we don't even get into the if statement here.  Does your fix to return `Unknown` account for this?  Said another way, does not entering this if statement still mean that the return statement below it is always correct?  I would have expected the fix to check if `g_fProcessDetach == true` and if so, just return `Unknown`.  I'll admit I'm not super familiar with the logic here which is why I ask.

Reply:
Yea I think that is a little better and simpler. There are other cases where `SendMethodEvent` is called without a code version where we would have returned `Optimized` (or `MinOptJit`) but those are kind of unknown cases, except for NGen methods. I have changed to the simpler way, it affects fewer things so the fix is more targeted as well.

Reply:
Excellent.  Thanks @kouvel.